src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -423,21 +422,16 @@ public void Abort(Exception error = null)
 
                 try
                 {
-                    ConnectionControl.End(ProduceEndType.SocketDisconnect);
+                    LifetimeControl.End(ProduceEndType.SocketDisconnect);
                 }
                 catch (Exception ex)
                 {
                     Log.LogError(0, ex, "Abort");
                 }
 
-                try
-                {
-                    RequestAbortedSource.Cancel();
-                }
-                catch (Exception ex)
-                {
-                    Log.LogError(0, ex, "Abort");
-                }
+                // Potentially calling user code. ThreadPool will catch and log any errors.
+                ServiceContext.ThreadPool.Run(() => RequestAbortedSource.Cancel());
```

Comment:
Dispatching user code or not is still selectable using InternalKestrelServerOptions